libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -872,14 +899,33 @@ class MapboxRouteLineApi(
                     .resourceProvider
                     .routeLineColorResources
                     .alternativeRouteUnknownCongestionColor,
-                routeLineOptions.resourceProvider.restrictedRoadSectionScale,
                 routeLineOptions.displaySoftGradientForTraffic,
                 routeLineOptions.softGradientTransition
             )
         } else {
             null
         }
 
+        // If the displayRestrictedRoadSections is true then produce a gradient that is transparent
+        // except for the restricted sections. If false produce a gradient for the restricted
+        // line layer that is completely transparent.
+        val restrictedRouteExpressionProducer = partitionedRoutes.first.firstOrNull()?.route?.run {
+            if (routeLineOptions.displayRestrictedRoadSections) {
+                MapboxRouteLineUtils.getRestrictedLineExpressionProducer(
+                    this,
+                    0.0,
+                    routeLineOptions.resourceProvider.routeLineColorResources.restrictedRoadColor,
+                    0,
```

Comment:
After the introduction of https://github.com/mapbox/mapbox-navigation-android/pull/4739, this assumption is not true anymore. I don't think we need to consider it as a blocker, but please cut a "know issue" ticket so that we can follow up.